mail/src/main/java/org/jboss/as/mail/extension/MailSessionAdd.java

```diff
@@ -90,19 +90,19 @@ protected void performRuntime(OperationContext context, ModelNode operation, Mod
         installRuntimeServices(context, address, fullTree);
     }
 
-    private static void addCredentialStoreReference(ServerConfig serverConfig, OperationContext context, ModelNode model, ServiceBuilder serviceBuilder, String... modelFilter) throws OperationFailedException {
+    private static void addCredentialStoreReference(ServerConfig serverConfig, OperationContext context, ModelNode model, ServiceBuilder serviceBuilder, final String modelFilter1, final String modelFilter2) throws OperationFailedException {
```

Comment:
I agree this method signature could be different; it does not need a variable argument.

From the point of view of the caller, what `modelFilter` should be is an argument to tell the method to where to find the credential-reference model node on the current model, passed by the third argument named `model`. 
Probably a PathAdress object, e.g `final PathAddress credRefParentAddress` is more suitable there instead of `String modelFilter1, final String modelFilter2`

Reply:
+1 but it can be a PathElement, which is cheaper.

Reply:
@yersan @bstansberry I can't change easily to a PathElement due to line [MailSessionAdd#L132](https://github.com/wildfly/wildfly/blob/24.0.0.Final/mail/src/main/java/org/jboss/as/mail/extension/MailSessionAdd.java#L132) . I have kept two Strings but renamed them to their content.

Reply:
Hi @boris-unckel , PathElement object has a (String,String) static constructor, see https://github.com/wildfly/wildfly-core/blob/main/controller/src/main/java/org/jboss/as/controller/PathElement.java#L58

Reply:
Hello @yersan thanks for the direct pointer. It's resolved now.